open_spiel/python/algorithms/external_sampling_mccfr.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2019 DeepMind Technologies Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Python implementation for Monte Carlo Counterfactual Regret Minimization."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import pyspiel
+
+# Indices in the information sets for the regrets and average policy sums.
+_REGRET_INDEX = 0
+_AVG_POLICY_INDEX = 1
+
+
+class ExternalSamplingSolver(object):
+    """
+    An implementation of external sampling MCCFR
+    """
+
+    def __init__(self, game, average_type='k_simple'):
```

Comment:
Can we make the average_type an enum defined at the top of this file?

Reply:
I am not sure I get it. Do you mean define at the top of this file something like:
```
from enum import Enum
average_type = Enum('average_type', 'k_simple k_full')
```
And then having it called in the constructor as default parameter by using:
`def __init__(self, game, average_type=average_type.k_simple.name)`?

Reply:
Yes but no worries I can change it upon import.

Reply:
Ok, will let it as it is then. Thanks